controllers/dockermachine_controller.go

```diff
@@ -47,17 +68,257 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 
 	dockerMachine := &infrastructurev1alpha2.DockerMachine{}
 	if err := r.Client.Get(ctx, req.NamespacedName, dockerMachine); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
 		log.Error(err, "failed to get dockerMachine")
+		return ctrl.Result{}, err
```

Comment:
can you file an issue that we don't know if we should be wrapping these errors with stack traces or not and as we use this our logging strategy will become obvious, see https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/65